src/network/networkd-routing-policy-rule.c

```diff
@@ -1052,6 +1064,15 @@ int manager_rtnl_process_rule(sd_netlink *rtnl, sd_netlink_message *message, Man
         if (r >= 0)
                 tmp->suppress_prefixlen = (int) suppress_prefixlen;
 
+        uint32_t suppress_ifgroup;
+        r = sd_netlink_message_read_u32(message, FRA_SUPPRESS_IFGROUP, &suppress_ifgroup);
+        if (r < 0 && r != -ENODATA) {
+                log_warning_errno(r, "rtnl: could not get FRA_SUPPRESS_IFGROUP attribute, ignoring: %m");
+                return 0;
+        }
+        if (r >= 0)
+                tmp->suppress_ifgroup = (int) suppress_ifgroup;
```

Comment:
Please check the range of the received value, and please use `int32_t` instead of `int`:
```
if (r >= 0) {
        if (suppress_ifgroup > INT32_MAX) {
                log_warning("rtnl: received invalid FRA_SUPPRESS_IFGROUP attribute value, ignoring: %"PRIu32, suppress_ifgroup);
                return 0;
        }
        tmp->suppress_ifgroup = (int32_t) suppress_ifgroup;
}
```
If you'd like, please add the similar check for `suppress_prefixlen`.